content/blog/using-jina-embeddings-haystack/index.md

```diff
@@ -0,0 +1,198 @@
+
+indexing_pipeline.run(data={"fetcher": {"urls": urls}})
+```
+
+## Building the query pipeline
+
+Now the real fun begins. Let's create a query pipeline so we can actually start asking questions. We write a prompt allowing us to pass our documents to the Mixtral-8x7B LLM. Then we initiatialize the LLM via the `HuggingFaceTGIGenerator`.
+
+In Haystack 2.0 `retriever`s are tightly coupled to `DocumentStores`. If we pass in the `retriever` we initialized earlier, this pipeline can access those embeddings we generated, and pass them to the LLM.
+
+```python
+
+from haystack.components.generators import HuggingFaceTGIGenerator
+from haystack.components.builders.prompt_builder import PromptBuilder
+
+from jina_haystack.text_embedder import JinaTextEmbedder
+prompt = """ Answer the question, based on the
+content in the documents. If you can't answer based on the documents, say so.
+
+Documents:
+{% for doc in documents %}
+  {{doc.content}}
+{% endfor %}
+
+question: {{question}}
+"""
+
+text_embedder = JinaTextEmbedder(api_key=jina_api_key, model_name="jina-embeddings-v2-base-en")
+generator = HuggingFaceTGIGenerator("mistralai/Mixtral-8x7B-Instruct-v0.1", token=hf_token)
+generator.warm_up()
+
+prompt_builder = PromptBuilder(template=prompt)
+query_pipeline = Pipeline()
+query_pipeline.add_component("text_embedder",text_embedder)
+query_pipeline.add_component(instance=prompt_builder, name="prompt_builder")
+query_pipeline.add_component("retriever", retriever)
+query_pipeline.add_component("generator", generator)
+
+query_pipeline.connect("text_embedder.embedding", "retriever.query_embedding")
+query_pipeline.connect("retriever.documents", "prompt_builder.documents")
+query_pipeline.connect("prompt_builder.prompt", "generator.prompt")
+```
+Time to ask a question!
+
+```python
+question = "Summarize what happened in Google v. Sonos"
+
+result = query_pipeline.run(data={"text_embedder":{"text": question},
+                                  "retriever": {"top_k": 3},
+                                  "prompt_builder":{"question": question},
+                                  "generator": {"generation_kwargs": {"max_new_tokens": 350}}})
+
+print(result['generator']['replies'][0])
+```
+```
+Answer: Google v. Sonos is a patent infringement case in which Sonos sued Google for infringing on two of its patents related to customizing and saving overlapping groups of smart speakers or other zone players according to a common theme..
+```
+
+## Exploring more questions and documents
+
+You can swap the `question` variable out and then call `pipeline.run` again:
+- What role did If This Then That play in Google v. Sonos?
+- What judge presided over Google v. Sonos?
+- What should Sonos have done differently?
+
+The indexing pipeline is written so that you can swap in other documents and analyze them. can You can try plugging the following URLs (or any PDF written in English) into the indexing pipeline and re-running all the code blocks below it.
```

Comment:
```suggestion
The indexing pipeline is written so that you can swap in other documents and analyze them. You can try plugging the following URLs (or any PDF written in English) into the indexing pipeline and re-running all the code blocks below it.
```